crates/extensions/tedge_gen_mapper/src/js_runtime.rs

```diff
@@ -97,6 +107,28 @@ impl JsRuntime {
         tracing::info!(target: "gen-mapper", "  - string count: {}", usage.str_count);
         tracing::info!(target: "gen-mapper", "  - atom count: {}", usage.atom_count);
     }
+
+    async fn send<Response>(
+        &self,
+        mut receiver: oneshot::Receiver<Response>,
+        request: JsRequest,
+    ) -> Result<Response, anyhow::Error> {
+        self.worker
+            .send(request)
+            .await
+            .map_err(|err| anyhow!(err))?;
+
+        // FIXME: The following timeout is not working
+        //  - see unit test: js_script::while_loop
+        //  - the issue is that the quickjs runtime fails to yield when executing `while(true)`
+        //  - Using task::spawn_blocking to launch the quickjs runtime doesn't help
+        //    - A timeout is the properly raised
+        //    - but the JS runtime keeps executing `while(true)` and is no more responsive.
```

Comment:
This is concern. We might have to:

- abort the whole JS runtime
- reload all the flows
- disable the faulty script and flow

Reply:
Fix by https://github.com/thin-edge/thin-edge.io/pull/3650/commits/8212eac9e3c23f52f496538b04e126bf4ba5a5c2